management/tests/commands/update-identity-from-user.test.ts

```diff
@@ -385,4 +390,81 @@ describe("Update identity from user", () => {
       );
     }
   });
+
+  test.only("should retry 2 times when marking the new identity fails with server error", async () => {
+    expect.assertions(9);
+
+    jest
+      .spyOn(fetchManagementConfig, "contextSetter")
+      .mockImplementation((managementCredentials: ManagementCredentials) => {
+        return (_, { headers }) => {
+          return {
+            headers: {
+              ...headers,
+              behaviour: "retry",
+              "targeted-failure": "mark",
+              "max-retry": "2",
+              authorization: `API_KEY ${managementCredentials.APIKey}`,
+            },
+          };
+        };
+      });
+
+    await updateIdentityFromUser(
+      mockedUpdateIdentityManagementCredentials,
+      "f3acadc9-4491-44c4-bd78-077a166751af",
+      "424242",
+      ["primaryEventId"],
+      primaryNewIdentity.value,
+      primaryIdentityToUpdate.id,
+    );
+
+    expect(spiedOnGetIdentity).toHaveBeenCalledTimes(3);
```

Comment:
```suggestion
    const maxRetry = 2
    jest
      .spyOn(fetchManagementConfig, "contextSetter")
      .mockImplementation((managementCredentials: ManagementCredentials) => {
        return (_, { headers }) => {
          return {
            headers: {
              ...headers,
              behaviour: "retry",
              "targeted-failure": "mark",
              "max-retry": maxRetry.toString(),
              authorization: `API_KEY ${managementCredentials.APIKey}`,
            },
          };
        };
      });

    await updateIdentityFromUser(
      mockedUpdateIdentityManagementCredentials,
      "f3acadc9-4491-44c4-bd78-077a166751af",
      "424242",
      ["primaryEventId"],
      primaryNewIdentity.value,
      primaryIdentityToUpdate.id,
    );

    expect(spiedOnGetIdentity).toHaveBeenCalledTimes(maxRetry + 1);
```
So we don't have to change it everywhere if we change the value of `maxRetry` (but its a detail)

Reply:
Very good point 👍